packages/angular_devkit/build_angular/src/babel/webpack-loader.ts

```diff
@@ -80,6 +85,21 @@ export default custom<AngularCustomOptions>(() => {
         shouldProcess = true;
       }
 
+      if (optimize) {
+        const angularPackage = /[\\\/]node_modules[\\\/]@angular[\\\/]/.test(this.resourcePath);
+        customOptions.optimize = {
+          // Angular packages provide additional tested side effects guarantees and can use
+          // otherwise unsafe optimizations.
+          looseEnums: angularPackage,
+          pureTopLevel: angularPackage,
+          // JavaScript modules that are marked as side effect free are considered to have
+          // no decorators that contain non-local effects.
+          wrapDecorators: !!this._module?.factoryMeta?.sideEffectFree,
```

Comment:
Previously we only ran BO, on known TypeScript libraries https://github.com/angular/angular-cli/blob/a41f2ed85c0f01be9a7a99d2fd40c495cd6248c3/packages/angular_devkit/build_optimizer/src/build-optimizer/webpack-plugin.ts#L28-L29

Should we retain this behaviour?

Reply:
That was an attempt to workaround the unsafe nature of the existing build optimizer passes.  With the increased safety and reduced number of AST mutations, the workaround can be removed.  In the event of any edge cases, additional strictness checks can be added to the new passes.  The typings field check also generates a large amount of false positives since many libraries contain typings but are not written in TypeScript.